### DISCUSSION TITLE
Prototype to Package

### DISCUSSION THREAD

**Discussion by SudoWodo** [SudoWodo/StochOptim#9] [Announcements]
Dear Mentors,

As per your suggestions package has been constructed form the prototype.
I am sure it is likely that the package is not set up properly thus I request your inputs and suggestions to correctly configure it.

## Steps to install the package
Here are the steps to setup the package locally on your machine.

1. Copy the repo on your machine
2. Open the folder stochoptim
3. Open StochOptim.Rproj
4. Run the following command in console to use it

```
library(devtools)
install()
```

5. Open vignettes
6. Open example.Rmd 
7. Execute the markdown or click Knit

**Comment by nashjc** [comment]
Unfortunately, in putting up the package, the soptcontrol.xls spreadsheet has been deleted. Also the CITATION example (which would need modification). 
I also noticed that my .gitignore was deleted. That should not be under version control as far as I am aware.

JN

**Reply by SudoWodo** [reply]
During setting up the package I was getting various error messages during R CMD checks so removed inst directory, NAMESPACE, NEWS for time being. But I have carefully saved these on my pc that includes CITATION and soptcontrol.xls . I am not sure about .gitignore though.

**Comment by nashjc** [comment]
Seems to be this commit that did the damage.

commit a086a604dbe8a952bebbc7128f4f02695074da43
Author: Siddhant Raghuvanshi ***@***.***>
commit 6202062daf2c1e04d5b89b46211ac4ea39c5af91 (HEAD -> main, origin/main, origin/HEAD)
Author: Siddhant Raghuvanshi ***@***.***>
Date:   Fri Jun 10 13:09:32 2022 +0530

     prototype to package

  .gitignore                          |   4 ++++
  ProjectDiary.txt                    |   3 +++
  stochoptim/.Rbuildignore            |   2 ++
  stochoptim/.gitignore               |   1 +
  stochoptim/DESCRIPTION              |   6 ++++++
  stochoptim/NAMESPACE                |  61 +++----------------------------------------------------------
  stochoptim/NEWS                     |   6 ------
  stochoptim/R/globalWrapper.R        |  75 +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
  stochoptim/R/localWrapper.R         |  78 ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
  stochoptim/R/optimizer_wrapper.R    | 164 
++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
  stochoptim/StochOptim.Rproj         |  20 ++++++++++++++++++++
  stochoptim/inst/CITATION            |   1 -
  stochoptim/inst/doc/soptcontrol.xls | Bin 28160 -> 0 bytes
  stochoptim/man/globalWrapper.Rd     |  29 +++++++++++++++++++++++++++++
  stochoptim/vignettes/.gitignore     |   2 ++
  stochoptim/vignettes/example.Rmd    |  92 
++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
  16 files changed, 479 insertions(+), 65 deletions(-)
